### ISSUE TITLE
new ctypes.pointer error when using subscriptable type annotations

### ISSUE THREAD

**Issue by nathanpage-credo** [microsoft/pyright#3823]
**Describe the bug**

ctypes.pointer is subscriptable for typing (if it is stringified) this is a mypy behavior as well. In the latest pylance release this is broken, meaning I can no longer give type definitions to ctypes pointer objects.

Returns error:

```log
Expected class type but received "(__arg: _CT@pointer, /) -> _Pointer[_CT@pointer]"
```

**To Reproduce**

Use type annoations with "ct.pointer[...]"

**Expected behavior**

No error should be given.

**Screenshots or Code**

```python
import ctypes as ct

class Foo(ct.Structure):
    ... # other stuff defined here 
    test: "ct.pointer[ct.c_int32]"  # these now error
    test2: "ct.pointer[ct.c_bool]"  # these now error
```

**VS Code extension or command-line**
Pylance 2022.8.20 introduces this issue

**Comment by erictraut** [comment]
This is due to a recent change in the typeshed stubs (in particular, `ctypes/__init__.pyi`). Mypy hasn't yet incorporated these recent typeshed changes, which is why you don't see the issue there.

If you think the change is incorrect, please file a bug report in the [typeshed issue tracker](https://github.com/python/typeshed).
